### ISSUE TITLE
New input method: Morse key

### ISSUE THREAD

**Issue by CrazyMarvin** [Crazy-Marvin/Morse#38]
This feedback came in:

> I used to be a radio ham 50 odd years ago and still use Morse in my electronic projects (search Deodorino). Lately I've been aware that a few of my acquaintances have unfortunately suffered strokes leaving them with speech problems. With this in mind, would it take much to add to your excellent project the facility to generate Morse by tapping the screen?
> [...]
> As for tapping the screen, we used to send Morse code by pressing a switch ( Morse 'key') which was connected to the transmitter. If you pressed the switch the transmitter was 'on' for as long as you pressed, so by pressing for short and long periods we sent the codes. On a mobile phone screen it would be necessary to switch on the screen, led, or speaker tone whenever there was finger contact. The sender adjusts the speed automatically in this case.
> [...]
> The amount of time the finger is in contact with the screen would determine whether a dot, dash, space etc was intended.

I think [this video](https://youtu.be/i8Ed0TSp0gs) shows it quite nicely.

**Comment by CrazyMarvin** [comment]
Here is a first mockup:

![TelegraphKey](https://user-images.githubusercontent.com/15004217/91352076-974ca400-e7e9-11ea-9321-ecb44e3a31db.PNG)

What do you think of it?

**Comment by CrazyMarvin** [comment]
This approach integrates the Telegraph directly into the keyboard:

![TelegraphKeyIntoKeyboard](https://user-images.githubusercontent.com/15004217/91770316-47068500-ebe1-11ea-9cb5-e7e4b5dc68a8.jpg)

**Comment by CrazyMarvin** [comment]
Telegraph key was added in release 1.6.

**Comment by sudomain** [comment]
Hello, I'm trying to use the telegraph key in 1.6 from FDroid. I've found where it is in the app, but it seems unresponsive or more likely I'm doing something wrong. In the button/area that says "tap for dit, hold for dah", am I supposed to be tapping/holding in that rectangle?

Also, it appears to be limited to typing in the Morse app itself. Are there plans to make it into a proper Android keyboard/input method so it can be used in other apps? I know of various keyboards that have two buttons (one for dits, one for dahs), but none that utilize the user's tempo as described in OP.

**Comment by CrazyMarvin** [comment]
> In the button/area that says "tap for dit, hold for dah", am I supposed to be tapping/holding in that rectangle?

Yes, that should be working. There will be a new release next year with some performance improvements though.
If it is not working, please upload a .gif of the behaviour so that we can investigate.


> Also, it appears to be limited to typing in the Morse app itself. Are there plans to make it into a proper Android keyboard/input method so it can be used in other apps? I know of various keyboards that have two buttons (one for dits, one for dahs), but none that utilize the user's tempo as described in OP.

No, there are no plans. Do you know how to do that? A PR would be very welcome.
